lib/aggregate/pipeline.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+module Aggregate
+  class Pipeline
+    attr_accessor :stages
+
+    def initialize(klass = nil)
+      @klass = klass
+      @stages = []
+    end
+
+    def execute
+      raise "Pipeline initializer must specify a class in order to be executable" if @klass.nil?
+
+      db = Mongoid.default_client.database
+      handler = Aggregate::Values::Pipeline.new(self)
+      command = "function(){return db.#{@klass.collection_name}.aggregate(#{handler.transpose}).toArray()}"
+
+      # https://github.com/mongodb/mongo-ruby-driver/blob/master/lib/mongo/operation/result.rb
+      result = db.command("$eval": command, nolock: true, await_data: true)
+      result.documents
+    end
+
+    Symbol.add_key(:and, "$and", "__multi__")
+    Symbol.add_key(:nor, "$nor", "__multi__")
+    Symbol.add_key(:or, "$or", "__multi__")
```

Comment:
These feel like they would be more appropriate in an initializer.

Reply:
Agreed